### PR TITLE
Rogue apls cata

### PR DESCRIPTION
**Pull request by Smufrik**
assasination and sub apl created by rewriting wowsims

### CODE REVIEW

**Comment by Smufrik** [comment]
i have fixed the issues you saw in the code before

**Review by PettyPeterPettigrew — APPROVED** [review]

**Comment by alexspew** [comment]
Small change needed, we don't want to use rupture during cleave/aoe, so only use rupture when our target is a boss or we only have one target.

actions.aoe+=/blade_flurry,if=active_enemies>1&!buff.blade_flurry.up
actions.aoe+=/fan_of_knives,if=active_enemies>3&energy<50&buff.deep_insight.up
actions.aoe+=/actions << you call normal actions here, which includes rupture
actions.aoe+=/cancel_buff,name=blade_flurry,if=active_enemies=1&buff.blade_flurry.up

actions+=/rupture,if=combo_points=5&!ticking&(boss|active_enemies==1)

**Review by Hekili — APPROVED** [review]

**Comment by rentatank** [comment]
I think you have missed updating the rogue.lua profiles at the bottom

**Comment by Smufrik** [comment]
What is it i have missed?

File: Cataclysm/Rogue.lua [hunk 1]
```diff
@@ -1251,7 +1288,57 @@ spec:RegisterAbilities( {
...
+        end,
+    },
+    -- Transferes any existiong combo points to the current enemy target
+    redirect = {
+        id = 73981,
+        range = 40,
+        cast = 0,
+        cooldown = 60,
+
+        handler = function ()
```
Comment (Hekili) [inline Cataclysm/Rogue.lua:1331]: This won't work.

File: Cataclysm/Rogue.lua [hunk 2]
```diff
@@ -1275,7 +1362,18 @@ spec:RegisterAbilities( {
         copy = { 2070, 6770, 11297, 51724 },
     },
 
-
+    smokeBomb = {
```
Comment (Hekili) [inline Cataclysm/Rogue.lua:1365]: `smoke_bomb`

File: Cataclysm/Rogue.lua [hunk 3]
```diff
@@ -1275,7 +1362,18 @@ spec:RegisterAbilities( {
...
+        id = 76577,
+        cast = 0,
+        cooldown = 180,
+        radius = 8,
+        duration = 5,
+        handler = function ()
+            local smokeCloud = CreateSmokeCloud(8, 5) -- Create a smoke cloud with a radius of 8 yards that lasts for 5 seconds
```
Comment (Hekili) [inline Cataclysm/Rogue.lua:1372]: This doesn't exist / isn't needed for addon purposes.

File: Cataclysm/Rogue.lua [hunk 4]
```diff
@@ -1331,7 +1429,7 @@ spec:RegisterAbilities( {
         texture = 135428,
 
         handler = function ()
-            -- TODO: Apply offhand poison.
+            dispelEnrageEffectsOnTarget()
```
Comment (Hekili) [inline Cataclysm/Rogue.lua:1432]: This function doesn't/shouldn't exist.  You can `removeDebuff( "target", "dispellable_enrage" )` if I'm remembering correctly, though.

File: Cataclysm/Rogue.lua [hunk 5]
```diff
@@ -1345,8 +1443,9 @@ spec:RegisterAbilities( {
         gcd = "totem",
 
         spend = function()
-            if talent.improved_sinister_strike.rank == 2 then return 40 end
-            if talent.improved_sinister_strike.rank == 1 then return 42 end
+            if talent.improved_sinister_strike.rank == 2 then return 39 end
+            if talent.improved_sinister_strike.rank == 2 then return 41 end
+            if talent.improved_sinister_strike.rank == 1 then return 43 end
             return 45
```
Comment (Hekili) [inline Cataclysm/Rogue.lua:1449]: `return 45 - ( 2 * talent.improved_sinister_strike.rank )`

**Comment by rentatank** [comment]
Your missing the spec:registerpack updates to match the APLs you made.

`spec:RegisterPack( "Assassination", 20240516, [[Hekili:TAvBVTTnq4FmfiVG1i)wR3wwsa6(Yqcg8aQsr)WqLiTeTeHPi1iPIRbc4V9DhLSLKR0YA2qacK5D85EUJhFognl6XOWuQLfTA(05lMoB(YG5tF)YflIcT7lzrHL0KT0m4djTa()hmgQXWLulxjDKl2P2zxNeKLDj66EHIMIqAuv6eW9CRT0C9KjhDd(Yk2ojraOCvwfpLzMq7I5vAvwfBsu46kUWEVmADhcoF20zlb0lzjrREhappnLv7cZKef(yo34iLAUsZT7De8xRPgwQJuvI81MZCKJKXr(SMAZDeprcGmqR2WfaVFZBCKts1pI8Y9aAz2sh5psSQ1mTJG0Y9G7bAc6NjOuZsufRP2F42jwnpzRjwTjgcCSvttzVLV52mTQQC4DuQ8rBqBglJka6oOXmQwRSE4nkHQdJaJBbA093gbpHftLPXPWh4EwxTztq)Ldsv7K9XHMQzsQGlzX6ktoUr04tSyMKvWzM7M1191ciHJ3iQ069VKVBOsSoTvcEyg25UURRkTvAp1TubtAdYRKzmD8gLoETqPsdaMUwWspZNAFJvm5olL5n2u7As42GC6MowOgeTEf4CkSubtK2HHQNy6TCHOpZoU6Py8evYRRXVU9)VQ5d8JjFIjvfOnSBsfxQ4sRjiLTHNWT3oToojkrAt22F)TgghIVJ4DZ8ZUO5CPXtFM98ZqNGoBFGLxazJkUG(1BoXpnRGYLMl7gTIklxqRVymu4UBAuiubnG7DLy2r1sUmZef(5p8Xv3V63U2rCKhr5dErPsBDe403rohGHwjSN7iA2FvX1OwJrva(rRSQciYWcj5uOHXe4E43H7ooY7b0(K0uvIiHouZxaUExdopaB6piQ5LMQ10k4OWeOF1ardzYWsepbK50C5wgKKiPV3wVjVyrbtMIb0MtHLzqMVVNGPmrubAPGjo01OVgia5kh5p)KHHiXkmF5TGcAopjVR3u5(2O6isfc(xlfyfw0IBk(vJk8HG(locW9MW8yneGe7xqqt7U08orEh0W3jJAG0EWvF1WVKSQauP9hkcLfQN3xCOMpVTSl4gSy9quiCOLR0W5)HzerHEJ(5K1h1WNR8ZnR3DuiQUgf2CPm6xJSqJeAVDLWeitHJgkowByH2w46BerBXOO1xK0rUdQCTiDICnc17ELq1vkhX59VsC61FJaTCuGEbDDh5my4(4QXO5H03X1TuDgZg0yUzwYjhdnRIC8h)Npnhm(TaDQzeXF6LY6t16BZ2E69D6AooXbX)N)Fh)6Pri2ZMok4(zlT7zGjqEaMnkadPq7i36itBjyVbrTbRziGpaJF9B8a0IuBa8Gn(TVHb7gV0cW2lo2a2DqMJ88ZG443mmZVXHhO5ixoCEo(T5HP2D9YZdJgbOS4BFBEm5r9T)dhXZ7Qpw)M2tuix01JMx2EIlJND4lCB3DZ1BFAu)x0F)d]] )

spec:RegisterPack( "Combat", 20240516, [[Hekili:Dw1Y2TTnq0VfTrBQRI1d300ALfnBI9cVHoztpbeqGdLWrKa8aaAfTbF7DgsjtqgkDS7cjboZCVdMh8k2C2ZSKmHhypT42flVDX85ZMVC5NUBblXFScyjvc5EXw8GwuIF)ft5gHNmFSWiYi4otTvIUyjBQvf(h0SnJZjgAfizpTILStLLbTbcojl55DkxGtFeb(Pug4MC8zPxz0bEHY5r35gBG)vyVQqnJL0yKUdcdG)8ut5aAXMciJ9pSePv5bRsGbG08cKcAOubipFoWNh4tB5p2X9b(Y2WnASIkezqAErT1EK5Xc69KGiEYf6utE6EngJJiA5aIu5tyj1vDiCfkjKk0zPz4bcYQMc9SBLgRDWM68w1EiMlm07OqL4OYKwzuAVBMeRaqJdU764aEr5KGLMu9G7PPsUOUW)Ap9mK9k5(b56YTKn155Z6xhZYmh0xRih2x6yBR1e3FWYwU3rnv)oi1BX50)NM0OPAsGxx1SDmwpmWxh4rTrBDLV22K9)4IuI7f2Th7OaxZwfXHidnlkuAi1w72rC9XR3xhGywCVz4w7F(EPQP2B8ftvVCGVbIi2M6QSa0U0GNKTYdx9vX32GCr8GSYCAJnE(D5vfNPW0bERWAn(27iElFbSoYER4085FKLCqy1yT4oRdvzn5QcOvqQu5W9OTbURUQYy9N0G2stuLmWX7VEpGRhbEG)GVfutNOe0zqggXobAgWeFK4wzWlkEsPLf1zKwbOWs3(xHhd8FpW)3V5aIjO09JBc8d7uYDXrl0h7YAGRne5)ScFvs5l64nJorjh6s6FJAQ2ZP55wkqHWFqKMfBAruMpGt6Ok6eL(ZH20nAmPRl3awkXUcJFw4XhkPggzy5aHC0joGQ97mwwY3Xfa63tT9Zto3SxxO(T1FyK1KBu5RBwMcpogI2LMX9DANGyGwwWwYRHH)vcgq8EpfvF59ppFAFd3VCib9u8hHHFbqFTqeXK1Xv2PG6ROn0DNI(nJPCT(oQqFfecG0ZJF(xUeRVKiEmQ30S5k3E01jruc2K6QPx623byGOfbSVg79R6f)Wb6Lu(61CIK4UcMPJQusInS)7p]] )

spec:RegisterPack( "Subtlety", 20240516, [[Hekili:LEvZYTTnq4NfFX5qDzLCTCCtJ8mPxATp4dHUtp0jKeKeseJibyXpsvD8WN9UaGuKecIYo(GtOaw8T)GD3VfrZJEokmhjXrpD9SRVz2I5Fmy2DZMp)MOq5(ACuynkBdAn8bfvb)BOkvwIL71BSVKHY1aiykEgSzuyQIukFGgL6h1RbzRXzrpbWxqYZXwjXISOWNliIMe9FOMKwT2KWwb)otsy0MKsIqcBVIXBs(d8gsjjamdoBfPuBzvQvCYMMKVYwRGt2zPnp28OfbXpS8NYqLLX2FgRX7kTBTKqjYZlfRgtX8RiRwkr81yzGKuHJLS4Cc((zxErotgWv1sfhdBLTHqxpeuUIEmMc5a7lqBhGKBrCckTeBfPfY4keHkH)UAlQuHxMXQszX1myrrqMIZXu5YfxMQwTkOcjKyEmBvSOnkeOQpJAWBjImmhU1Ivc80k5IUJg4ACNZxWzQARsCChJDRJVqGLKfdcY2I57dYz7ohORi084Dy0gkwiIr11L7hc6weLikGaWlVy(TOabGgNJOzytyPhE7nmOajhU)e6aOSaUI5OCShPeLKmCmc0Eo8HoXWF8YZrRz6fS2V97dkJvgVLW9Pojgvkl8HghxHZjsKPqXZjnUmC(ApBIQsvcFO2E3(M8RcCfJZl01UhVzkuvlKO0tH4NV5DeKTz(JL2j7P3ZoKg(gXBC6Iqook9kQkmh5yx682UvxJS7E1DCvLZbHeKAe3j9WSZKz6wRDq9IwTMI09DbQ7VBMx51jB9E24sUl9E5V4yBUpP(auNTAEeg2C7rblpDlCouFoS(G5yJE6x03ftFZZr6ACpvNZ0vnmP2VWUxqNWbkHgzx1)V1mbK6WHJQpLalLa7JiOlDi2kH3a)9lb2yOxRaW7aH9TrH7qCQgKOW)6lF9PhE63)utstYZfa3kPQMXLTuXFq3u(dnjC8)OiCCEtIGvbcHuswf46Wczfi6ASiO5rZXxXklz700JGqCeWPVdZH1HyeimK8Mi1IzDrlVFtsQs2jhLz0TIos68CTWWyhOuKa)jOqn5hBsoMsrxbBh2OD2b7uhveHWyscv9aVBTgasgOeoHUbdHnDu4bP9qAmRQW0CTHllqWYyJoaSjmobg)aSWSsfmJdSfbSuENL93)jKsaiHReF7kW1kizfdLgr33R1wNgUgHMeezzpUdJbDk9xHbN4DQ5zlenjZ)Mg08HlD9anVJuwoWJAHu2jQjAywIQQsX2l6syKNMhFWKoOx4gNP1GnJcnFPNt02)f(6jZCNwjHbnpUfuuiMQZZZJ(TijKuQfVFLWmWjHBfe8LNe6MKLnjl6XFCFvnE)SdEwg4U)3HjUhOwXK65xhy)T8Yo28IHImQxMJG3ocRdnpDK6JdLY2wZrI7ExbPwklnq)Yqv13oYrDZN9g13NH0JEC7AOzGA(BekTP3KCPUPGFEZPU8N)9KnDsTPQhedpWdd6rQFyZkKQuEiHFGo7oI7lnSvlX2hBPBSoD6)XpePj5(MKzgZ9cOB4Xpj5vQ72Avpfl9zm0tEAH0gbmEGh3NbrnGKQhSokt9xWK7VURcVpZXGvRz4ohMVG572uGW8jh8BOXmEyaFr2ZBkt9kPro(XteEOT1BxHhM5Qj5Lx6kdgp31qL7zclB2GWBUqF28eHrpTPoDvX3txb37cFwLNGQ)A)X8ddg(EkoIZtdE7jTXXZKB6bC3mxwf7L1bYKjcDoZuRZYpzp9fEjVMKp61Nu5N)SJJA6RlFPHUmOtZJ5F0FVCJtZIDIhe0Ju)69muhXu6W)E6m2l6McZ9zdN00DBm0J1PEkXejf3VCet)WhNy7duBgYmfPBhywaEQqbSz4x(p4zi2HcKrr)p]])
`

**Pull request merged**